java/code/src/com/suse/manager/matcher/MatcherJsonIO.java

```diff
@@ -182,7 +182,9 @@ public List<SystemJson> getJsonSystems(String arch, boolean includeSelf, boolean
                 boolean virtualHost = entitlements.contains(EntitlementManager.VIRTUALIZATION_ENTITLED) ||
                         !system.getGuests().isEmpty();
                 boolean countVCores = !virtualHost && system.getInstalledProductSet()
-                        .map(s -> s.getBaseProduct().getChannelFamily()).stream()
+                        .map(productSet -> productSet.getBaseProduct())
+                        .map(baseProduct -> baseProduct.getChannelFamily())
+                        .stream()
                         .anyMatch(cf -> vCoreCountedChannelFamilies.contains(cf.getLabel()));
```

Comment:
We can make this more readable. `Stream` usage is unnecessary and a bit misleading here since we're dealing with a single object.

```suggestion
                        .map(cf -> vCoreCountedChannelFamilies.contains(cf.getLabel()))
                        .orElse(false); // Case when any of the mappings above return empty.
```